product_code_builder_sequence/models/product_product.py

```diff
@@ -46,3 +46,18 @@ def copy(self, default=None):
                     'prefix_code': self.prefix_code + _('-copy'),
                 })
         return super(ProductTemplate, self).copy(default)
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def copy(self, default=None):
+        for product in self:
+            if default is None:
+                default = {}
+            if product.prefix_code:
```

Comment:
@StefanRijnhart , Implemented the check for an already existing prefix_code in default, in both copy methods. Thanks for the hint.

@gdgellatly , You're right: product.prefix_code was morphing in self.prefix_code. This never caused troubles as copy methods are supposed to run on a single record, therefore self and product are, in most cases, the same object. 
I've fixed this mistake in the latest commit. Thanks for your review.